app/backend/approaches/chatreadretrievereadvision.py

```diff
@@ -145,15 +145,13 @@ async def run_until_final_call(
                 )
                 vectors.append(vector)
 
-        # Only keep the text query if the retrieval mode uses text, otherwise drop it
-        if not has_text:
-            query_text = None
-
         results = await self.search(
             top,
             query_text,
             filter,
             vectors,
+            use_text_search,
+            use_vector_search,
```

Comment:
We weren't passing it because we relied on vectors being an empty list versus a filled list. I could have kept with that design, since we only go through effort of populating vectors when we're using them, but it felt weird to have a "use_text_search" boolean and not have a "use_vector_search" boolean. So I added both options for symmetry/API consistency.